packages/react/src/use-feature-flag.ts

```diff
@@ -1,20 +1,62 @@
-import { Client, EvaluationDetails, FlagValue, ProviderEvents } from '@openfeature/web-sdk';
-import { useEffect, useState } from 'react';
+import { Client, EvaluationDetails, FlagEvaluationOptions, FlagValue, ProviderEvents, ProviderStatus } from '@openfeature/web-sdk';
+import { Dispatch, SetStateAction, useEffect, useState } from 'react';
 import { useOpenFeatureClient } from './provider';
 
-export function useFeatureFlag<T extends FlagValue>(flagKey: string, defaultValue: T): EvaluationDetails<T> {
-  const [, setForceUpdateState] = useState({});
+type ReactFlagEvaluationOptions = {
+  /**
+   * Suspend flag evaluations while the provider is not ready.
+   * Set to false if you don't want to use React Suspense API.
+   * Defaults to true.
+   */
+  suspend?: boolean,
+  /**
+   * Update the component if the the provider emits a change event.
+   * Set to false if you never want to update components live, even if flag value changes
+   * are received by the associated provider.
+   * Defaults to true.
+   */
+  updateOnChange?: boolean,
+} & FlagEvaluationOptions;
 
+const DEFAULT_OPTIONS: ReactFlagEvaluationOptions = {
+  suspend: true
+};
+
+enum SuspendState {
+  Pending,
+  Success,
+  Error
+}
+
+/**
+ * Evaluates a feature flag, returning evaluation details.
+ * @param {string}flagKey the flag identifier
+ * @param {T} defaultValue the default value
+ * @param {ReactFlagEvaluationOptions} options options for this evaluation
+ * @template T flag type
+ * @returns { EvaluationDetails<T>} a EvaluationDetails object for this evaluation
+ */
+export function useFeatureFlag<T extends FlagValue>(flagKey: string, defaultValue: T, options?: ReactFlagEvaluationOptions): EvaluationDetails<T> {
+  const defaultedOptions = { ...DEFAULT_OPTIONS, ...options };
+  const [, updateState] = useState<object | undefined>();
+  const forceUpdate = () => updateState({});
   const client = useOpenFeatureClient();
 
   useEffect(() => {
-    const forceUpdate = () => setForceUpdateState({});
 
-    // adding handlers here means that an update is triggered, which leads to the change directly reflecting in the UI
-    client.addHandler(ProviderEvents.Ready, forceUpdate);
-    client.addHandler(ProviderEvents.ConfigurationChanged, forceUpdate);
+    if (client.providerStatus !== ProviderStatus.READY) {
+      client.addHandler(ProviderEvents.Ready, forceUpdate); // update the UI when the provider is ready
+      if (defaultedOptions.suspend) {
+        suspend(client, updateState);
+      }
+    }
```

Comment:
This is a good point, and I blocker I think. We may need some options or something. I will think more about it.

Reply:
I think we just have a problem with the SDK implementation.
https://openfeature.dev/specification/sections/events#requirement-533 tells that all handlers of an associated state must run also if they are attached after the provider reached this specific state.
The SDK currently only does this for the `READY` state. When we change/fix this, the problem would be gone as we could also just listen for error events. @toddbaert 
https://github.com/open-feature/js-sdk/blob/main/packages/client/src/client/open-feature-client.ts#L52-L64

Reply:
I will fix this now.

Reply:
https://github.com/open-feature/js-sdk/issues/700

Reply:
Yes, I think for now it's OK. Of course if we decide we need different error semantics, we can always change things since this SDK is experimental for now.

Reply:
Sounds good!

Reply:
fix: https://github.com/open-feature/js-sdk/pull/701